playground/src/Playground.tsx

```diff
@@ -40,30 +53,55 @@ type PlaygroundProps = {
 /**
   Used to render your component inside a live-editable playground and directly see the output of the code used.
 */
-export const Playground = ({ code, scope, noInline }: PlaygroundProps) => {
+export const Playground = ({
+  useShadowDOM,
+  styles,
+  code,
+  scope,
+  noInline,
+}: PlaygroundProps) => {
   const commentsRegex = /([^http:|https:]\/\/.*)|(\/\*(.|\n)*?\*\/)/gm;
-  return (
+  const template = (
     <LiveProvider
       code={code.trim()}
       scope={scope}
       noInline={noInline}
       theme={oceanicNext}
       transformCode={(code) => {
         const codeWithoutComments = code.replaceAll(commentsRegex, '').trim();
-        return codeWithoutComments.startsWith('<') ? `<>${codeWithoutComments}</>` : codeWithoutComments;
+        return codeWithoutComments.startsWith('<')
+          ? `<>${codeWithoutComments}</>`
+          : codeWithoutComments;
       }}
     >
       <div>
-        <div style={styles.preview} id="playground-preview">
+        <div style={playgroundStyles.preview}>
           <LivePreview />
         </div>
-        <LiveEditor style={styles.editor} />
-        <LiveError style={styles.error} />
+        <LiveEditor style={playgroundStyles.editor} />
+        <LiveError style={playgroundStyles.error} />
       </div>
     </LiveProvider>
   );
+
+  if (!useShadowDOM) {
+    return template;
+  }
+
+  const playgroundRef = useRef();
+
+  useEffect(() => {
+    const playgroundElement = playgroundRef.current as DockitReactPlayground;
+    if (!playgroundElement) return;
```

Comment:
another issue I had here is that `playgroundElement` can be not defined
never run into this myself and kinda missing the point of it, maybe React gurus can help

and TS was also not happy here, even with a guard `if (!playgroundElement)`, but `as DockitReactPlayground` was needed anyway and solved it

Reply:
AFAIK in `useEffect ` it's safe to assume it's defined, so I'll remove this protective style line of code